src/Tooltip2.tsx

```diff
@@ -0,0 +1,141 @@
+  direction?: 'n' | 'ne' | 'e' | 'se' | 's' | 'sw' | 'w' | 'nw'
+  text?: string
+  noDelay?: boolean
+  align?: 'left' | 'right'
+  wrap?: boolean
+  type?: 'label' | 'description'
+} & SxProp
+
+export type TriggerPropsType = {
+  'aria-describedby'?: string
+  'aria-labelledby'?: string
+  'aria-label'?: string
+}
+
+const tooltipClasses = ({direction, noDelay, align, wrap}: Omit<Tooltip2Props, 'type' | 'text' | 'children'>) => ({
+  position: 'relative',
+  display: 'inline-block',
+  '& > span': {
+    position: 'absolute',
+    zIndex: '1000001',
+    display: 'none',
+    padding: '0.5em 0.75em',
+    fontSize: 0,
+    color: 'fg.onEmphasis',
+    textAlign: 'center',
+    textDecoration: 'none',
+    textShadow: 'none',
+    textTransform: 'none',
+    letterSpacing: 'normal',
+    wordWrap: 'break-word',
+    whiteSpace: 'pre',
+    pointerEvents: 'none',
+    backgroundColor: 'neutral.emphasisPlus',
+    borderRadius: '3px', // use theme value radii.1'
+    opacity: 0,
+  },
+  '&::before': {
+    position: 'absolute',
+    zIndex: '1000001',
+    display: 'none',
+    content: '""',
+    width: 0,
+    height: 0,
+    color: 'neutral.onEmphasis',
+    pointerEvents: 'none',
+    borderStyle: 'solid',
+    borderWidth: '6px',
+    borderColor: 'transparent',
+    opacity: 0,
+  },
+  '&:hover, &:focus, &:active, &:focus-within': {
+    '& > span': {
+      display: 'inline-block',
+      opacity: 1,
+      //   conditionally render styles depending on direction
+      ...((direction === 'n' || direction === 'ne' || direction === 'nw') && {
+        right: '50%',
+        bottom: '100%',
+        marginBottom: '6px',
+      }),
+      // only for n direction
+      ...(direction === 'n' && {
+        transform: 'translateX(50%)',
+      }),
+      ...((direction === 's' || direction === 'se' || direction === 'sw') && {}),
+    },
+    '&::before': {
+      display: 'inline-block',
+      textDecoration: 'none',
+      opacity: 1,
+      //   conditionally render styles depending on direction
+      ...((direction === 'n' || direction === 'ne' || direction === 'nw') && {
+        borderTopColor: 'neutral.emphasisPlus',
+        top: '-7px',
+        bottom: 'auto',
+        right: '50%',
+        marginRight: '-6px',
+      }),
+    },
+  },
+})
+
+const Tooltip2: React.FC<React.PropsWithChildren<Tooltip2Props>> = ({
+  direction = 'n',
+  text,
+  noDelay,
+  align,
+  wrap,
+  sx = {},
+  children,
+  type = 'label',
+  ...props
+}) => {
+  const id = useId()
+  const tooltipRef = useRef<HTMLDivElement>(null)
+  const child = Children.only(children) // make sure there is only one child
+
+  // we need this check for every render
+  useEffect(() => {
+    if (tooltipRef.current) {
+      const childNode = tooltipRef.current.children[0] // For now, I assume it has one node but that is not true
+      if (!isFocusable(childNode as HTMLElement)) {
+        throw new Error('Tooltip2: The child element must be focusable')
+      }
```

Comment:
@TylerJDev 
>Would it be possible to enhance this through TypeScript, or log a warning on dev to ensure the trigger element (child) is an interactive element?

I added a runtime check - not sure if that would be possible with the type check. I chat this with @joshblack as well. Let me know what you think about that!

Reply:
Oh wow, I love this! So basically we're checking the element that's passed to `Tooltip2`, and seeing if it's focusable with the `isFocusable` util? 

This is perfect, and I love how slick it is! Let's definitely run with this :grin:

Reply:
>Oh wow, I love this! So basically we're checking the element that's passed to Tooltip2, and seeing if it's focusable with the isFocusable util?

Yes 🙌🏻